src/ui/topeditorcontainer.cpp

```diff
@@ -82,10 +82,17 @@ EditorTabWidget *TopEditorContainer::tabWidgetFromEditor(Editor *editor)
 
 void TopEditorContainer::on_currentTabChanged(int index)
 {
+    if(index == -1)
+        return;
+
     EditorTabWidget *tabWidget = dynamic_cast<EditorTabWidget *>(sender());
     if (!tabWidget)
         return;
 
+    if( m_currentTabWidget == tabWidget && m_currentTabWidget->currentIndex() == index) {
```

Comment:
This fixes the crash on moving windows. Moving a tab triggers the onCurrentChanged signal to be emitted every time the tab switches position with another one while dragging it across the tab bar.

If that happens, something somewhere chokes up and JS signals aren't handled fast enough. This can sometimes lead to the QEventLoop not having finished its exec() yet before another signal should be emitted. And since that QEventLoop is now a member function and calling exec() twice, everything goes boom.